psi4/src/psi4/libmints/writer.cc

```diff
@@ -1417,6 +1417,36 @@ void MOWriter::write_mos(Molecule & mol){
 
     CharacterTable ct = mol.point_group()->char_table();
 
+    BasisSet& basisset = *wavefunction_->basisset().get();
+    
+    std::vector<std::string> ao_labels;
+    for (int s = 0; s < basisset.nshell(); s++) {
+        GaussianShell shell = basisset.shell(s);
+        int center = shell.ncenter()+1;
+        int am = shell.am();
+        char amchar = shell.amchar();
+        std::string basename = mol.symbol(shell.ncenter()) + std::to_string(center) + " ";
+        basename += char(amchar);
+        if (amchar == 'd' && (basisset.has_puream())) {
+            for (int i = 1; i < 6; i++) ao_labels.push_back(basename+" "+std::to_string(i));
```

Comment:
w.r.t @loriab's comments it would probably be a good trade-off between keeping the printing consistent with qchem, readability, and making the output easier to parse if `basename + " "+std::to_string(i)` was simply changed to `basename + "_"+std::to_string(i)` here. 

Thoughts?